math/perfect_number.cpp

```diff
@@ -0,0 +1,24 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
```

Comment:
Separate the code into documented functions and structures if needed as well as add self-test cases using `assert` (requires the `cassert` library/header).